core/src/parking_lot.rs

```diff
@@ -1035,3 +1042,67 @@ unsafe fn unpark_filter_internal(key: usize,
 
     result
 }
+
+/// Register an resource identified by key in the deadlock detector
+#[cfg(feature = "deadlock_detection")]
+pub fn register_resource(key: usize) {
+    unsafe {
+        let mut thread_data = None;
+        let thread_data = get_thread_data(&mut thread_data);
+        (*thread_data.resources.get()).push(key);
+    }
+}
+
+/// Deregister an resource identified by key in the deadlock detector
+#[cfg(feature = "deadlock_detection")]
+pub fn deregister_resource(key: usize) {
+    unsafe {
+        let mut thread_data = None;
+        let thread_data = get_thread_data(&mut thread_data);
+        let resources = &mut (*thread_data.resources.get());
+        match resources.iter().rposition(|x| *x == key) {
+            Some(p) => resources.swap_remove(p),
+            None => {
+                check_deadlock();
```

Comment:
Why do you call `check_deadlock` here? This code can only be reached if there is a bug in the implementation of `Mutex` (or whatever primitive calls `deregister_resource`).

Reply:
It's a debug artifact, I'll remove it.